pkg/diff/sql_generator.go

```diff
@@ -1461,20 +1476,20 @@ func (csg *checkConstraintSQLGenerator) Add(con schema.CheckConstraint) ([]State
 
 	if !con.IsValid {
 		sb.WriteString(" NOT VALID")
-	} else if !csg.isNewTable {
```

Comment:
We strip the migration hazards for new tables within the TableSQLGenerator, so we don't actually need this conditional